Bucketeer/Sources/Internal/Evaluation/EvaluationStorageImpl.swift

```diff
@@ -65,36 +65,36 @@ final class EvaluationStorageImpl: EvaluationStorage {
             try evaluationDao.deleteAll(userId: userId)
             try evaluationDao.put(userId: userId, evaluations: evaluations)
         }
-        // Update cache directly after we have force update
+        // Update cache directly
         evaluationMemCacheDao.set(key: userId, value: evaluations)
         self.evaluatedAt = evaluatedAt
     }
 
     func update(evaluations: [Evaluation], archivedFeatureIds: [String], evaluatedAt: String) throws -> Bool {
-        var activeEvaluations = [Evaluation]()
-        var archivedEvaluationIds = [String]()
-        try evaluationDao.startTransaction {
-            // 1. Update evaluation with new data
-            try evaluationDao.put(userId: userId, evaluations: evaluations)
-            // 2. Get current data in db
-            let evaluationsInDb = try evaluationDao.get(userId: userId)
-            // 3. Filter active & archived
-            for evaluation in evaluationsInDb {
-                if archivedFeatureIds.contains(evaluation.featureId) {
-                    archivedEvaluationIds.append(evaluation.id)
-                } else {
-                    activeEvaluations.append(evaluation)
-                }
-            }
-            // 4. Remove all the evaluations which have the same id in the list archivedEvaluationIds
-            if (archivedEvaluationIds.count > 0) {
-                try evaluationDao.deleteByIds(archivedEvaluationIds)
+        // 1. Get current data in db
+        var activeEvaluationsDict = try evaluationDao.get(userId: userId)
+            .reduce([String:Evaluation]()) { (input, evaluation) -> [String:Evaluation] in
+                var output = input
+                output[evaluation.featureId] = evaluation
+                return output
             }
+        // 2. Update evaluation with new data
+        for evaluation in evaluations {
+            activeEvaluationsDict[evaluation.featureId] = evaluation
         }
-        self.evaluatedAt = evaluatedAt
-        // 5. Save a new cache
-        evaluationMemCacheDao.set(key: userId, value: activeEvaluations)
-        return evaluations.count > 0 || archivedEvaluationIds.count > 0
+        // 3. Filter active
+        let activeEvaluations = activeEvaluationsDict.values.filter { evaluation in
+            !archivedFeatureIds.contains(evaluation.featureId)
+        }.map { item in
+            item
+        }.sorted { evaluation1, evaluation2 in
```

Comment:
Just confirming, do we need to sort it out?

Reply:
yes, because the dictionary used to filter | update the evaluation list is un-order. when we read the current evaluations from the dictionary, the list will be un-order... sometime the test will fail. [EvaluationStorageTests.swift](https://github.com/bucketeer-io/ios-client-sdk/pull/44/files#diff-2b6ab159d1c204385cbcc02ffe6c632ab790bbf329111192f5a6b04ea7722607)

There is an order-dictionary from the swift-collections library.  
https://github.com/apple/swift-collections/blob/main/Documentation/OrderedDictionary.md
But I don't want to add more dependencies to our framework. and sorting is much simple.

If you think we need find another way more effective, let me know.

Reply:
I see.

How about changing the test to check if it contains the test items?

Reply:
let me changes the tests

Reply:
I removed the sorting logic and update the test 
https://github.com/bucketeer-io/ios-client-sdk/pull/44/commits/23be1e074284375659583c3d9c48b9d9a27a37d6